### DISCUSSION TITLE
Cannot install ocaml-lsp?

### DISCUSSION THREAD

**Discussion by novemberkilo** [mason-org/mason.nvim#1350] [General]
Hi - I cannot install ocaml-lsp via Mason. I am reasonably sure that the issue is not with the ocamllsp server (as I am able to install and use that locally with vscode) and I'm curious to know if the issue I am running into is something local to Mason.

Here's what I am doing:
1. `:MasonInstall ocaml-lsp`
2. I see Mason attempting to do this: `opam install --destdir=. --yes --verbose ocaml-lsp-server.1.10.2` 
3. This fails with `spawn: opam failed with exit code 20 and signal 0.`

Some observations:
1. In my local opam switch I have `ocaml-lsp-server` installed. I would have expected Mason to just detect this and not try and install it again.
2. The version of  ocaml-lsp-server is now `1.15.1` -- is there a configuration issue that is causing Mason to try and install an older version? Also, is this perhaps the cause for the failure?

**Comment by williamboman** [comment]
Hey! There's an issue template for package installation issues that asks for more information, please use that one if you're still having issues :pray:!

> * In my local opam switch I have `ocaml-lsp-server` installed. I would have expected Mason to just detect this and not try and install it again.

Hm, why would you `:MasonInstall` it if you already have it installed and prefer to use that installation?



> 2\. The version of  ocaml-lsp-server is now `1.15.1` -- is there a configuration issue that is causing Mason to try and install an older version? Also, is this perhaps the cause for the failure?

Package versions are kept up to date via Renovate in https://github.com/mason-org/mason-registry. Unfortunately Renovate doesn't support Opam packages, so these will be have to be updated manually (PRs always welcome!).

**Reply by williamboman** [reply]
I've gone ahead and updated the two opam packages that exists in mason here:
- https://github.com/mason-org/mason-registry/pull/1852
- https://github.com/mason-org/mason-registry/pull/1853

Maybe the version upgrade fixes your issues.

**Reply by novemberkilo** [reply]
Thanks for the quick updates @williamboman - I can see that `Mason` has picked up the updates but continue to get the `spawn: opam failed with exit code 20 and signal 0.` error. I'll see about opening an issue using the template you mentioned. Thanks.

**Reply by Phamier** [reply]
@novemberkilo Same here

**Reply by Trzyq0712** [reply]
I've had the same issue. Instead, you can manually install the appropriate ocaml lsp server (using opam for example - `opam install ocaml-lsp-server`) and just skip using Mason in this case, i.e. use lspconfig directly.
